### PR TITLE
Initial Cypress config for RBAC

### PR DESCRIPTION
**Pull request by radekkaluzik**
https://issues.redhat.com/browse/RHCLOUD-28830

### CODE REVIEW

File: config/webpack.cy.config.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    cacheDirectory: path.resolve(__dirname, '../.cypress-cache'),
+  },
+  stats: {
+    errorDetails: true,
+  },
+  plugins: [
+    new MiniCssExtractPlugin({
+      filename: '[name].[contenthash].css',
+    }),
+    new ModuleFederationPlugin({
```
Comment (Hyperkid123) [inline config/webpack.cy.config.js:83]: Not needed

File: config/webpack.cy.config.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      {
+        test: /\.(jpe?g|svg|png|gif|ico|eot|ttf|woff2?)(\?v=\d+\.\d+\.\d+)?$/i,
+        type: 'asset/resource',
+      },
+    ],
+  },
+  resolve: {
+    extensions: ['.tsx', '.ts', '.js'],
+    alias: {
+      ...searchIgnoredStyles(path.resolve(__dirname, '../')),
```
Comment (Hyperkid123) [inline config/webpack.cy.config.js:61]: Not needed

File: config/webpack.cy.config.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        test: /\.(js|ts)x?$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'swc-loader',
+          options: {
+            jsc: {
+              experimental: {
+                plugins: [
+                  [
+                    'swc-plugin-coverage-instrument',
```
Comment (Hyperkid123) [inline config/webpack.cy.config.js:21]: I don't think you have this installed

File: cypress.config.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { defineConfig } from 'cypress';
+const { addMatchImageSnapshotPlugin } = require('@simonsmith/cypress-image-snapshot/plugin');
+
+export default defineConfig({
+  component: {
+    specPattern: 'cypress/component/**/*.cy.{js,jsx,ts,tsx}',
+    excludeSpecPattern: ['/snapshots/*', '/image_snapshots/*', '/src/*'],
+    setupNodeEvents(on, config) {
+      addMatchImageSnapshotPlugin(on, config);
```
Comment (Hyperkid123) [inline cypress.config.ts:10]: I don't think we want snapshots

File: cypress/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# Visual regression testing
```
Comment (Hyperkid123) [inline cypress/README.md:1]: This does not exist here.

File: cypress/fixtures/helpTopicsResponse.json [hunk 6]
```diff
@@ -0,0 +1,151 @@
+{
```
Comment (Hyperkid123) [inline cypress/fixtures/helpTopicsResponse.json:1]: This file is not needed.

File: cypress/fixtures/settings-navigation.json [hunk 7]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (Hyperkid123) [inline cypress/fixtures/settings-navigation.json:1]: This file is not needed.

File: cypress/fixtures/testUser.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (Hyperkid123) [inline cypress/fixtures/testUser.json:1]: This file is not needed.

File: cypress/support/component.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+import { mount } from 'cypress/react18'
+import '@cypress/code-coverage/support'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
+
+const { addMatchImageSnapshotCommand,  } = require('@simonsmith/cypress-image-snapshot/command');
+
+// configure the matcher to be less precise to adjust between the headless and normal browser rendering differences
+addMatchImageSnapshotCommand({
```
Comment (Hyperkid123) [inline cypress/support/component.ts:32]: THis config can be removed

File: package.json [hunk 10]
```diff
@@ -84,6 +84,7 @@
     "@redhat-cloud-services/eslint-config-redhat-cloud-services": "^2.0.3",
     "@redhat-cloud-services/frontend-components-config": "^6.0.5",
     "@redhat-cloud-services/tsc-transform-imports": "^1.0.4",
+    "@simonsmith/cypress-image-snapshot": "^8.1.2",
```
Comment (Hyperkid123) [inline package.json:87]: This can be removed.

File: scripts/e2e.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+/* eslint-disable @typescript-eslint/no-var-requires */
+const { spawn, execSync } = require('child_process');
+const waitOn = require('wait-on');
+
+const options = {
+  resources: ['https://localhost:1337/webpack-dev-server'],
+  delay: 6000,
+  interval: 3000, // wait for 3 sec
+  validateStatus: function (status) {
+    console.log({ status });
```
Comment (Hyperkid123) [inline scripts/e2e.js:11]: console log

**Comment by Hyperkid123** [comment]
Snyk warning is fine, it yells at test files.

File: package.json (hunk 10 above)
Comment (radekkaluzik) [reply package.json:87]: removed

File: scripts/e2e.js (hunk 11 above)
Comment (radekkaluzik) [reply scripts/e2e.js:11]: removed

File: cypress/support/component.ts (hunk 9 above)
Comment (radekkaluzik) [reply cypress/support/component.ts:32]: removed

File: cypress/fixtures/testUser.json (hunk 8 above)
Comment (radekkaluzik) [reply cypress/fixtures/testUser.json:1]: File removed.

File: cypress/fixtures/settings-navigation.json (hunk 7 above)
Comment (radekkaluzik) [reply cypress/fixtures/settings-navigation.json:1]: File removed.

File: cypress/fixtures/helpTopicsResponse.json (hunk 6 above)
Comment (radekkaluzik) [reply cypress/fixtures/helpTopicsResponse.json:1]: File removed.

File: config/webpack.cy.config.js (hunk 3 above)
Comment (radekkaluzik) [reply config/webpack.cy.config.js:21]: removed

File: config/webpack.cy.config.js (hunk 1 above)
Comment (radekkaluzik) [reply config/webpack.cy.config.js:83]: Removed

File: config/webpack.cy.config.js (hunk 2 above)
Comment (radekkaluzik) [reply config/webpack.cy.config.js:61]: removed

File: cypress.config.ts (hunk 4 above)
Comment (radekkaluzik) [reply cypress.config.ts:10]: removed

File: cypress/README.md (hunk 5 above)
Comment (radekkaluzik) [reply cypress/README.md:1]: removed

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**